### PR TITLE
feat: Introduce RISC-V support

### PR DESCRIPTION
**Pull request by SimonIT**
Not tested yet, love to hear some comments

EDIT: Works flawless

### CODE REVIEW

**Comment by Frosty-J** [comment]
Looks alright, though I won't be able to test it with the hardware I currently have. I might get a RISC-V SBC at some point, though don't see much reason to get one over ARM at this time.

I wouldn't worry about 32-bit RISC-V (but worth supporting if it's trivial to). The `is64Bit` boolean isn't very ready for 128-bit RISC-V, but I don't know if any 128-bit hardware even exists yet (it's in the specification).

While I think about it, there's a lot of natives downloaded and bundled into libGDX game JARs (x86, x64, arm32, arm64... potentially RISC-V in future – Windows, macOS, Linux) – maybe an opt-in solution could be realised for that? E.g. if someone is only interested in 64-bit Windows (not unheard of for Steam titles), don't bother with the rest. But this is probably a discussion for elsewhere, and not the most important thing in the world.

**Comment by Berstanio** [comment]
> I agree with you about choosing the architecture. Maybe it's something for the gdx-setup app? Or do we add Gradle tasks that generate one jar per architecture?

I'm not sure, whether this isn't a bit bloated. We already have 6 jars produced with natives, if now desktop gets split up too we end up with like around 23 jars if we split on `os` and `arch`. Only splitting on `arch` would be around 11 jars, which is still a lot in my opinion.

**Comment by PokeMMO** [comment]
At first glance, we should consider replacing `isARM` and `isRISCV` with a more generic arch enum (or some other extendable argument)

While there, it might also make sense to turn `is64` into an enum of potential bitness.

I think this should all be able to do be done while ensuring backwards compatibility too.

**Comment by SimonIT** [comment]
Done, do you think we should deprecate methods with `boolean is64Bit, boolean isARM`? I also think the extra handling for android and ios a bit ugly. Should we also change the shared library loader?

**Comment by PokeMMO** [comment]
> Done, do you think we should deprecate methods with `boolean is64Bit, boolean isARM`? 

Yeah, the legacy methods should be marked as deprecated.

> Those constants are then pretty useless:

I was assuming we could replace them with references to the enums in a backwards compatible manner, but I haven't experimented to check if that's actually possible.

>Should we also change the shared library loader?

I think so, but doing so would be very breaking, and might be worthwhile keeping the old ones as accurate as we can.

**Comment by SimonIT** [comment]
Done, except for the shared library loader. The reason is that I'm not sure where to put the enums because, in the current position, it's not able to access them

**Comment by SimonIT** [comment]
Do you have an idea? Should we move the enum to `SharedLibraryLoader`? Or do we want to leave it as it is?

**Comment by PokeMMO** [comment]
I think move it to SharedLibraryLoader, and make a new static field for the arch. Also leave the old isARM, isRISCV, is64Bit but mark them as deprecated.

**Comment by SimonIT** [comment]
Everything moved

File: gdx-jnigen-gradle/src/main/java/com/badlogic/gdx/jnigen/gradle/JnigenExtension.java [hunk 1]
```diff
@@ -82,36 +86,52 @@ public void all(Action<BuildTarget> container) {
...
 
-	public void add(TargetOs type, boolean is64Bit, boolean isARM) {
+	@Deprecated
+	public void add(Os type, boolean is64Bit, boolean isARM) {
 		add(type, is64Bit, isARM, null);
 	}
 
-	public void add(TargetOs type, Action<BuildTarget> container) {
-		add(type, false, false, container);
+	public void add(Os type, Action<BuildTarget> container) {
```
Comment (Berstanio) [inline gdx-jnigen-gradle/src/main/java/com/badlogic/gdx/jnigen/gradle/JnigenExtension.java:112]: The following method override is missing for backwards compatibility:
```java
	public void add(Os type, Architecture.Bitness bitness, Action<BuildTarget> container) {
		add(type, bitness, Architecture.x86, container);
	}
```

Comment (SimonIT) [reply gdx-jnigen-gradle/src/main/java/com/badlogic/gdx/jnigen/gradle/JnigenExtension.java:112]: Well noticed!

File: gdx-jnigen-loader/src/main/java/com/badlogic/gdx/utils/SharedLibraryLoader.java [hunk 2]
```diff
@@ -36,34 +36,60 @@
...
+	@Deprecated
+	static public boolean isMac = os == Os.MacOsX;
+	@Deprecated
+	static public boolean isIos = os == Os.IOS;
+	@Deprecated
+	static public boolean isAndroid = os == Os.Android;
+	@Deprecated
+	static public boolean isARM = architecture == Architecture.ARM;
+	@Deprecated
+	static public boolean isRISCV = architecture == Architecture.RISCV;
```
[the remark this answers is not in the document]
Comment (SimonIT) [reply gdx-jnigen-loader/src/main/java/com/badlogic/gdx/utils/SharedLibraryLoader.java:89]: Yeah, just forgot to remove it

File: gdx-jnigen/src/main/java/com/badlogic/gdx/jnigen/BuildTarget.java [hunk 3]
```diff
@@ -91,9 +89,7 @@ public enum TargetOs {
 	public String minIOSVersion = "11.0";
 
 	/** Creates a new build target. See members of this class for a description of the parameters. */
-	public BuildTarget (BuildTarget.TargetOs targetType, boolean is64Bit, String[] cIncludes, String[] cExcludes,
-		String[] cppIncludes, String[] cppExcludes, String[] headerDirs, String compilerPrefix, String cFlags, String cppFlags,
-		String linkerFlags) {
+	public BuildTarget(Os targetType, Architecture.Bitness bitness, String[] cIncludes, String[] cExcludes, String[] cppIncludes, String[] cppExcludes, String[] headerDirs, String compilerPrefix, String cFlags, String cppFlags, String linkerFlags) {
```
[the remark this answers is not in the document]
Comment (SimonIT) [reply gdx-jnigen/src/main/java/com/badlogic/gdx/jnigen/BuildTarget.java:92]: I have no real opinion on this

**Comment by PokeMMO** [comment]
Merged!

Next step should be PR with `-SNAPSHOT` to libgdx to triple check before we make a new release.

**Pull request merged**